src/training/final.ts

```diff
@@ -27,13 +27,82 @@ type Coupon = {
   amount: number;
 };
 
+const coupon_100: Coupon = { couponId: 1, couponName: "coupon_100",amount: 100 };
+const coupon_500: Coupon = { couponId: 2, couponName: "coupon_500", amount: 500 };
+const coupon_1000: Coupon = { couponId: 3, couponName: "coupon_1000", amount: 1000 };
+
 type Wallet = {
-  walletId;
-  username;
-  balance;
-  method;
-  transactionHistroy;
-  coupon;
+  walletId: number;
+  username: string;
+  balance: number;
+  method: 0 | 1 | 2;
+  transactionHistroy: Transaction[];
+  coupon?: Coupon[];
+};
+
+type Transaction = {
+   walletId: number;
+   paymentAmount: number;
+   transactionType: string;
+   method: 0 | 1 | 2;
+   transactionDate: Date;
+   balance: number;
 };
 
-type Transaction = {};
+export const userWallet: Wallet[] = [
+   {walletId:1, username: "田中一郎", balance: 10000, method: 1, transactionHistroy: []},
+   {walletId:2, username: "田中二郎", balance: 10000, method: 0, transactionHistroy: [], coupon: [coupon_100]},
+   {walletId:3, username: "田中三郎", balance: 10000, method: 2, transactionHistroy: [], coupon: [coupon_100, coupon_500]},
+   {walletId:4, username: "田中四郎", balance: 10000, method: 1, transactionHistroy: [], coupon: [coupon_100, coupon_1000]},
+   {walletId:5, username: "田中五郎", balance: 10000, method: 0, transactionHistroy: [], coupon: [coupon_100, coupon_500,coupon_1000]}
+];
+
+export const doSettlement = (walletId: number, paymentAmount: number): void => {
+
+   // 処理対象のウォレットのインデックスを所得する。
+   function getwalletId(): number {
```

Comment:
indexをとるので、命名を変えてください。
今回はindexとidが同じですが、異なる場合があるので
```suggestion
   function getIndex(): number {
```

Reply:
関数名をgetIndexに変更しました。